### PR TITLE
fix(942160): check REQUEST_FILENAME

### PR DESCRIPTION
**Pull request by mat1010**
This change addresses  #3779

### CODE REVIEW

File: tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942160.yaml
Comment (theseion) [inline tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942160.yaml:174]: Please also add a negative test, to show why we need `REQUEST_FILENAME`.

File: tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942160.yaml
Comment (theseion) [inline tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942160.yaml:186]:
```suggestion
          version: HTTP/1.1
```

File: tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942160.yaml
Comment (theseion) [inline tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942160.yaml:1]: You can add your name to the `authors` field, if you like.

**Review by theseion — CHANGES_REQUESTED** [review]

File: tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942160.yaml
Comment (mat1010) [reply tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942160.yaml:174]: Not sure if I understood this correctly. I added some negative tests with invalid sleep and benchmark syntax, that do not match. This change does not exactly shoud why we would need `REQUEST_FILENAME` but proves that it does not match broader than before.

Comment (theseion) [reply tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942160.yaml:174]: You're right, I didn't look closely enough.

**Review by theseion — APPROVED** [review]
Looks good. I'd like to get a second opinion before I merge though.

**Comment by dune73** [comment]
It's a nice bypass and a decent fix. Thanks @mat1010.

I wonder whether we still want to do `REQUEST_BASENAME` since we now cover `REQUEST_FILENAME`? It's redundant in most of the cases. But maybe `REQUEST_FILENAME` allows for a different type of bypass ...

**Comment by azurit** [comment]
While i was reviewing without reading @dune73's [comment](https://github.com/coreruleset/coreruleset/pull/3782#issuecomment-2262376084) i come to the same conclusion - `REQUEST_BASENAME` is not needed.

**Comment by mat1010** [comment]
You are right. `REQUEST_BASENAME` is redundant. I removed it in [f563dca](https://github.com/coreruleset/coreruleset/commit/f563dca4cf280303296c2daae7b468e200ba67bc). 

> But maybe REQUEST_FILENAME allows for a different type of bypass ...

Not sure if it allows some kind of bypass that wouldn't have worked before. In worst case it could catch more requests as the initial rule since it is now checking the full URI instead of just the name of the script.


While thinking about bypassing this rule I came to an additional easy bypass we have to consider. 

Using comments like `sleep(/*comment*/5)` or `pg_sleep(/*comment*/5)`- this syntax is not blocked, but would work on a mysql and posgresql server. For benchmark this wont work since the regex is just matching for everything before and after the `,` 
```
mysql> SELECT sleep(/*comment*/5);
+-----------+
| sleep( 5) |
+-----------+
|         0 |
+-----------+
1 row in set (5.01 sec)
```
```
postgres=# select pg_sleep(/*comment*/5);
 pg_sleep
----------

(1 row)
```

I will also work on a fix to specifically address comments in the `sleep` command

**Comment by theseion** [comment]
Try using `removeCommentsChar` or `replaceComments` before making the regular expression unnecessarily complex.

**Comment by mat1010** [comment]
Sorry for the long delay. I added the the comment handling and a test for it.

**Review by theseion — APPROVED** [review]

**Pull request merged**
